docs/docs.md

```diff
@@ -0,0 +1,581 @@
+      }
+    | {
+        // 2018orwil
+        event: string
+        // qm1
+        match: string
+        attempted: boolean
+        succeeded: boolean
+        statName: string
+      })[]
+}[]
+```
+
+# `/event/{eventKey}/team/{team}/stats/teleop`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = (
+  | {
+      // 2018orwil
+      event: string
+      // qm1
+      match: string
+      attempts: number
+      successes: number
+      statName: string
+    }
+  | {
+      // 2018orwil
+      event: string
+      // qm1
+      match: string
+      attempted: boolean
+      succeeded: boolean
+      statName: string
+    })[]
+```
+
+# `/event/{eventKey}/teams`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = string[]
+```
+
+# `/events`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  name: string
+  district?: string
+  week?: number
+  // UTC date
+  startDate: string
+  // UTC date
+  endDate: string
+  location: {
+    lat: number
+    lon: number
+  }
+}[]
+```
+
+# `/schema`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  teleop: {
+    statName: string
+    statKey: string
+    type: "number" | "boolean"
+  }[]
+  auto: {
+    statName: string
+    statKey: string
+    type: "number" | "boolean"
+  }[]
+}
+```
+
+# `/team/{team}/automodes`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = string[]
+```
+
+# `/team/{team}/stats/auto`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  modeName: string
+  stats: (
+    | {
+        // 2018orwil
+        event: string
+        // qm1
+        match: string
+        attempts: number
+        successes: number
+        statName: string
+      }
+    | {
+        // 2018orwil
+        event: string
+        // qm1
+        match: string
+        attempted: boolean
+        succeeded: boolean
+        statName: string
+      })[]
+}[]
+```
+
+# `/team/{team}/stats/teleop`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  team: string
+  teleop: (
+    | {
+        attempts: {
+          max: number
+          avg: number
+        }
+        successes: {
+          max: number
+          avg: number
+        }
+        statName: string
+      }
+    | {
+        // total
+        attempts: number
+        // total
+        successes: number
+        statName: string
+      })[]
+  auto: (
+    | {
+        attempts: {
+          max: number
+          avg: number
+        }
+        successes: {
+          max: number
+          avg: number
+        }
+        statName: string
+      }
+    | {
+        // total
+        attempts: number
+        // total
+        successes: number
+        statName: string
+      })[]
+}
+```
+
+# `/users/{userId}/stars`
+
+## `GET`
+
+Anyone can view anyone's stars
+
+### Response
+
+```ts
+type Data = string[]
+```
+
+# `/users/{userId}`
+
+## `GET`
+
+Anyone can view any user
+
+### Response
+
+```ts
+type Data = {
+  username: string
+  name: string
+  admin?: true
+}
+```
+
+
+## `PUT`
+
+Only admins can modify other users
```

Comment:
except users can modify themselves

Reply:
Right, that was my intent when writing this

Reply:
Seems vague to me.

Reply:
These comments are not very explicit. I think a better doc would be something like: "Admins can modify any property of any user expect for their ID. Users can modify themselves but not change whether they are an admin or whether they are verified. P.S. this needs verified.